### PR TITLE
(2272) Read and write to the new organisations - professions relation

### PR DESCRIPTION
**Pull request by pezholio**
This builds on #502 to move all the professions - organisation logic from the `organisation` and `additionalOrganisation` fields to the new `ProfessionToOrganisation` one to many entity and removes the old relations. 

I've had to dispese with my usual "all commits must be green approach" as I wanted to keep the commit history readable (hopefully I've done that!). Sorry there's a lot here, but it really needs to be shipped as one feature.

Once this is done, we can update the UI so up to five organisations can be added, and have a role applied to each.

### CODE REVIEW

File: src/professions/admin/top-level-information.controller.spec.ts [hunk 1]
```diff
@@ -102,11 +102,16 @@ describe('TopLevelInformationController', () => {
...
-          organisation: organisation,
-          additionalOrganisation: additionalOrganisation,
-        });
+        const profession = professionFactory.build(
+          {
+            name: 'Example Profession',
+          },
+          {
+            transient: {
+              organisations: [organisation, additionalOrganisation],
```
Comment (Gweaton) [inline src/professions/admin/top-level-information.controller.spec.ts:111]: Are these transient params doing anything here? Are they supposed to be from another commit?

File: src/professions/admin/presenters/organisations-presenter.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    this.organisations = organisations || [];
+  }
+
+  public list(): string {
+    if (this.organisations.length === 0) {
+      return '';
+    }
+
+    const organisations = this.organisations
+      .map((organisation) => organisation.name)
```
Comment (Gweaton) [inline src/professions/admin/presenters/organisations-presenter.ts:16]: Possibly me being overly paranoid here, but do we need to `escape` the name here, as we're outputting HTML?

File: src/testutils/factories/profession.ts [hunk 3]
```diff
@@ -1,12 +1,20 @@
...
 import { RegulationType } from '../../professions/profession-version.entity';
+import { ProfessionToOrganisation } from '../../professions/profession-to-organisation.entity';
+
 import industryFactory from './industry';
 import organisationFactory from './organisation';
 import qualificationFactory from './qualification';
 import userFactory from './user';
 
-class ProfessionFactory extends Factory<Profession> {
+type ProfessionTransientParams = {
```
Comment (Gweaton) [inline src/testutils/factories/profession.ts:13]: Ah, I think this is the commit it should belong to. I can see why it's a tricky one to get in order though...

**Review by Gweaton — APPROVED** [review]
Thanks for dealing with all this - an enormous effort. Have noticed there are 2 "Update changelog" commits here and a couple of tiny things.

As I'm sure you're aware, this was a very large PR so I found it a bit tricky to review. I hope I haven't missed anything - let's make sure we test this properly on staging before prod to make sure all's working as expected.

File: src/professions/admin/presenters/organisations-presenter.ts (hunk 2 above)
Comment (pezholio) [reply src/professions/admin/presenters/organisations-presenter.ts]: Yeah, good call - have done 👍

File: src/testutils/factories/profession.ts (hunk 3 above)
Comment (pezholio) [reply src/testutils/factories/profession.ts:13]: Yeah, it's a bit of a head wrecker alright! Ended up doing it all in one big WIP commit and then breaking it down after the fact. Spent too much time chasing down failing tests!
